### PR TITLE
ci: add lint/format/style checking

### PR DESCRIPTION
**Pull request by robbkidd**
Adds Rubocop to the project and then addresses its findings.

While this is a big diff, if you walk the commits, each has a message about a distinct lint/format/style change that was made.

### CODE REVIEW

**Comment by robbkidd** [comment]
I should have added this while working on #5 right when the Ruby was beginning to be written to avoid all these appeasement commits afterwards.

**Review by codeboten — APPROVED** [review]
This looks generally ok to me, not a ruby expert though :)

File: Makefile [hunk 1]
```diff
@@ -10,6 +10,22 @@ test:
 	@echo "+++ Running tests"
 	bundle exec rake spec
 
+#: run the lint checks
+lint:
```
Comment (MikeGoldsmith) [inline Makefile:14]: Should we all the missing `.PHONY` entries for the makefile targets that don't target files?

File: examples/hello-world/app.rb [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-Tracer = OpenTelemetry.tracer_provider.tracer('hello_world_tracer', '0.1.0')
+Tracer = OpenTelemetry.tracer_provider.tracer("hello_world_tracer", "0.1.0")
 
 def hello_world
   "Hello, World!"
     .tap do |message|
-      Tracer.in_span('hello') do |span|
-        Tracer.in_span('world') do |span|
-          span.set_attribute("message", message)
+      Tracer.in_span("hello") do |_hello_span|
```
Comment (MikeGoldsmith) [inline examples/hello-world/app.rb:20]: Is it normal to use a `_` prefix for variables that don't get used?

File: lib/honeycomb/opentelemetry/trace/baggage_span_processor.rb [hunk 3]
```diff
@@ -52,20 +54,23 @@ def on_start(span, parent_context)
...
-        # been exported.
+        # Export all ended spans to the configured `Exporter` that have not yet been exported.
+        # NO-OP method to satisfy the SpanProcessor duck type.
+        # Always successful; this processor does not maintain any state to flush.
         #
-        # @param [optional Numeric] timeout An optional timeout in seconds.
+        # @param [optional Numeric] timeout An optional timeout in seconds, unused in this implementation.
         # @return [Integer] 0 for success and there is nothing to flush so always successful.
-        def force_flush(timeout: nil)
+        def force_flush(_timeout: nil)
```
Comment (MikeGoldsmith) [inline lib/honeycomb/opentelemetry/trace/baggage_span_processor.rb:63]: Same here for `_timeout` - is it normal to use a`_` prefix?

**Review by MikeGoldsmith — APPROVED** [review]
Looks good to me - left some non-blocking queries.

**Comment by MikeGoldsmith** [comment]
We're not going ahead with ruby distro.

**Pull request closed without merging**